### ISSUE TITLE
Deleting an connection which is used in integration is still possible in 1.3.5

### ISSUE THREAD

**Issue by rhuss** [syndesisio/syndesis#2582]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Documentation issue or request
</code></pre>

Actually I just delete a connection which is used in an integration and no one stopped me. Now I have an instable system.

<img width="1428" alt="image" src="https://user-images.githubusercontent.com/99080/39836017-e87ca02c-5386-11e8-8698-5d9141a726ac.png">

## Tasks involved / Steps to Reproduce

1.
2.
3.
4.

**Comment by gashcrumb** [comment]
Yeah, we didn't fix this in #2524, what was fixed was the result of deleting a connection that is in use.  I think the issue was how to deal with bad imports etc.  But I think we should keep this issue open and decide how to deal with all these cases, we have many cases where objects keep references to other objects, we need to handle how these relations are maintained consistently.

**Comment by dongniwang** [comment]
Just wanted to chime in on this, I think it's also related to deleted connections. 

I was using staging site this morning and saw this on the integration list view - A random "?" appeared in the running/stopped column and the "Configuration Required" is not aglined with the rest in that column. 

<img width="1206" alt="screen shot 2018-07-16 at 1 17 38 pm" src="https://user-images.githubusercontent.com/24943812/42774261-452013a6-88ff-11e8-9fe0-9e33672e7c9d.png">

The "?" also appears on the integration details view.
<img width="636" alt="screen shot 2018-07-16 at 11 30 29 am" src="https://user-images.githubusercontent.com/24943812/42774336-6f3c9cfe-88ff-11e8-8dbb-204b474a7575.png">

Also related to this, @michael-coker found that importing an integration that uses connections that have been deleted gives you a broken import screen as well... 

cc: @gashcrumb @rhuss

**Comment by michael-coker** [comment]
When I import an integration that uses connections I've deleted from Syndesis, this is the import screen I see. It's worth noting the integration used (2) Amazon S3 connections (not twitter). Although it does seem to import the integration OK, and importing also restored the deleted connection.

![screen shot 2018-07-16 at 1 03 25 pm](https://user-images.githubusercontent.com/35148959/42774962-c9d957f8-88f8-11e8-86a7-6854d0e9ccf0.png)

**Comment by gashcrumb** [comment]
@michael-coker can you raise a separate issue about the import screen?  When an integration is exported I think it also exports all related items which include connections, so an import should restore those deleted connections.

I can certainly look at preventing the user from deleting connections that are in use.  Wonder, right now we've no information on how many integrations a connection is used in, like we do with API clients and extensions.  Should we have this?

**Comment by gashcrumb** [comment]
@zregvart wonder, would it be difficult to add a field to the Connection object that tracks how many integrations it's used by?  Otherwise I'll have to fetch Integrations as well on the connection list/detail pages to figure out if they can be deleted or not :-P

**Comment by gashcrumb** [comment]
We need #3098 implemented first for this.

**Comment by zregvart** [comment]
> @zregvart wonder, would it be difficult to add a field to the Connection object that tracks how many integrations it's used by? Otherwise I'll have to fetch Integrations as well on the connection list/detail pages to figure out if they can be deleted or not :-P

I don't think it's difficult at all, we do something similar for Connectors IIRC.

I think we only need to decide if we need this for 7.1.

Any thoughts @paoloantinori @heiko-braun?

**Comment by paoloantinori** [comment]
According to the discussion here I don't see it as a high priority that influences other bugs, so unless you guys point me to why this should get extra attention at this later stage, my call would be to delay it to the next version.
